sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventService.java

```diff
@@ -375,6 +368,18 @@ public Predicate createUserFilter(
 			eventParticipantJoin = eventParticipantPath;
 		}
 
+		final User currentUser = getCurrentUser();
+		final JurisdictionLevel jurisdictionLevel = currentUser.getCalculatedJurisdictionLevel();
+		if (jurisdictionLevel == JurisdictionLevel.NATION || currentUser.hasAnyUserRole(UserRole.REST_USER)) {
+			if (currentUser.getLimitedDisease() != null) {
+				return cb.or(cb.equal(eventJoin.get(Event.DISEASE), currentUser.getLimitedDisease()), cb.isNull(eventJoin.get(Event.DISEASE)));
```

Comment:
Calling return in the mid of a method can easily lead to unwanted behaviour. It should only be called at the top, related to validations checks for the method call or at the end, containing the result.

In this setting I'd suggest to do it the same we are doing it in CaseService.createUserFilter: For users without jurisidiction, we simply keep the Predicate filter unchanged (null). The limited disease check is done for all users below the jurisdiction check and uses CriteriaBuilderHelper.and.